wandb/apis/public/runs.py

```diff
@@ -782,17 +782,24 @@ def _full_history(self, samples=500, stream="default"):
         return [json.loads(line) for line in response["project"]["run"][node]]
 
     @normalize_exceptions
-    def files(self, names=None, per_page=50):
-        """Return a file path for each file named.
+    def files(self, names=None, pattern=None, per_page=50):
+        """Returns a `Files` object for all files in the run which match the given criteria.
+
+        You can specify a list of exact file names to match, or a pattern to match against.
+        If both are provided, the pattern will be ignored.
```

Comment:
Let's raise a ValueError instead to alert the user that their code isn't working as they might expect.